injector.py

```diff
@@ -1,56 +1,36 @@
-from pywinauto.handleprops import processid
+import ctypes
+import os
 from pywinauto.handleprops import is64bitprocess
 from pywinauto.timings import TimeoutError as WaitError
 from pywinauto import sysinfo
-from socket import socket
-from socket import AF_INET
-from socket import SOCK_DGRAM
-import cfuncs
-import ctypes
-import os
 
-# Relative path, would be changed after adding to pywinauto
-dll_path = "./dll_to_inject/"
+from . import cfuncs
+
 
 class Injector(object):
 
-    """Class for injections dll and set hook on windows messages"""
+    """Class for injections dll"""
 
-    def __init__(self, app, is_unicode = False):
-        """Constructor inject dll, set socket and hook (one application - one class instanse)"""
-        self.app = app
+    def __init__(self, pid, backend_name, dll_name, is_unicode = False):
+        """Constructor inject dll (one application - one class instanse)"""
         self.is_unicode = is_unicode
-        self.pid = processid(self.app.handle)
+        self.pid = pid
         if not sysinfo.is_x64_Python() == is64bitprocess(self.pid):
             raise RuntimeError("Application and Python must be both 32-bit or both 64-bit")
         self.h_process = self._get_process_handle(self.pid)
-        self.dll_path = os.path.abspath("{0}pywinmsg{1}{2}.dll".format(dll_path,
-            "64" if is64bitprocess(self.pid) else "32",
-            "u" if self.is_unicode else ""))
-        self._inject_dll_to_process()
-
-        self.sock = socket(AF_INET, SOCK_DGRAM)
-        self.sock.bind(('',0))
-        port = self.sock.getsockname()[1]
-
-        self._remote_call_int_param_func("InitSocket", port)
-        self._remote_call_void_func("SetMsgHook")
 
-    @property
-    def socket(self):
-        """Return datagram socket"""
-        return self.sock
-
-    @property
-    def application(self):
-        """Return hooked application"""
-        return self.app
+        self.dll_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                     'backends', backend_name,
+                                     'bin', 'x{}'.format("64" if is64bitprocess(self.pid) else "86"),
+                                     '{}{}.dll'.format(dll_name, "u" if self.is_unicode else "")).encode()
```

Comment:
`.encode()` to which encoding? Can it be more explicit? What is the default encoding?

Reply:
`locale.getpreferredencoding()` for Ascii functions and UTF-16 for W-functions